### DISCUSSION TITLE
Simple question on continuous node process

### DISCUSSION THREAD

**Discussion by Dutch-TS-Dev** [vercel/next.js#47384] [Help]
### Summary

Hey all,

I have a very simple NodeJS process (it makes a simple API call that pulls in a few KB or data every 2 seconds) 
This process needs to run 24/7. As far as I know Vercel only offers Serverless solutions for serverside apps. 

My question is : is Vercel suitable for this? I already have a NextJS in production and I would love to do as much as I can inside of Vercel, but it just seems to me that the way Vercel works is not suitable for a continous script like this one...

Kind regards

**Comment by guitheengineer** [comment]
Why vercel wouldn't be suitable for this case?

**Comment by icyJoseph** [comment]
Hi,

You can use CRON jobs in Vercel. https://vercel.com/guides/how-to-setup-cron-jobs-on-vercel

> I have a very simple NodeJS process (it makes a simple API call that pulls in a few KB or data every 2 seconds)

And I guess you save that data to disk, and use to serve future requests? If so, that is, you need to write to disk every 2 seconds, to serve future requests, then I think you might have an issue. 

- https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-temporary-storage
- https://github.com/orgs/vercel/discussions/241#discussioncomment-363002
- https://vercel.com/guides/is-sqlite-supported-in-vercel#local-storage-and-serverless

**Comment by Dutch-TS-Dev** [comment]
Thanks for the answers everybody. I tried cron jobs but they don't seem to work for me. here is what I get:
<img width="1326" alt="Screenshot 2023-03-24 at 10 12 11" src="https://user-images.githubusercontent.com/19747468/227414856-69a62a60-29fe-48c8-a0be-b5489d6fb66c.png">

I would expect it to start yesterday which it did not. 
^ also when I click on run here nothing happens. 
Furthermore it feels like abuse of what crons are intended to be: a periodically performed task. the nature of my job is continuous...I'm considering moving this process to heroku

**Reply by icyJoseph** [reply]
Yeah for continuous stuff it might be better to have a dedicated server. I know you said Node.js, but I have been wanting to try out Shuttle, who I've heard calls themselves, the Vercel for pure backend stuff - though they use Rust at the moment. https://docs.shuttle.rs/introduction/idle-projects they have this idea of idle projects, maybe that's something to look at? A good excuse to learn Rust and said you "rewrote it in Rust"?
